### ISSUE TITLE
Error when fetching hitran isotopes with no lines in the spectral range

### ISSUE THREAD

**Issue by CorentinGrimaldi** [radis/radis#248]
### 🐛 Describe the bug

Error raised when calculating a H2O spectrum with hitran.

### 🖥 Steps To Reproduce

```
from radis import calc_spectrum

spec_h2o = calc_spectrum(wavelength_min=4197,
                         wavelength_max=4254,
                         Tgas=300,
                         pressure=1.01325,
                         mole_fraction=0.001,
                         molecule='H2O',
                         isotope='1,2,3,4,5',
                         path_length=1,
                         databank='hitran',
                         )
```
Raises the error:
```
Traceback (most recent call last):

  File "C:\Users\Corentin Grimaldi\Documents\Python Scripts\BUG_ID_H5.py", line 10, in <module>
    spec_h2o = calc_spectrum(wavelength_min=4197,

  File "c:\users\corentin grimaldi\documents\python scripts\radis\radis\lbl\calc.py", line 414, in calc_spectrum
    _calc_spectrum(

  File "c:\users\corentin grimaldi\documents\python scripts\radis\radis\lbl\calc.py", line 588, in _calc_spectrum
    sf.fetch_databank(**conditions)

  File "c:\users\corentin grimaldi\documents\python scripts\radis\radis\lbl\loader.py", line 999, in fetch_databank
    self._fetch_molecular_parameters(df)

  File "c:\users\corentin grimaldi\documents\python scripts\radis\radis\lbl\loader.py", line 2179, in _fetch_molecular_parameters
    id_set = df.id.unique()

  File "C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\generic.py", line 5465, in __getattr__
    return object.__getattribute__(self, name)

AttributeError: 'DataFrame' object has no attribute 'id'
```

### 💡 Possible solutions

No idea, maybe something change with the DataFrame object ? Because I already have done these simulations in the past without any errors...

**Comment by erwanp** [comment]
Thank you for posting this, I can reproduce the bug on 0.9.30.

### Explanation

Weird : by using `verbose=3` I got the reason : 
```
The following columns had the `object` format and were removed: ['id', 'iso', 'globu', 'globl', 'locu', 'locl', 'lmix']
```

So there is non-numeric value in the ìd` column of the loaded database (more on that at the end)
Non-numeric values are removed, for performance, by the "drop_non_numeric=True" argument of [fetch_databank()](https://radis.readthedocs.io/en/latest/source/radis.lbl.loader.html#radis.lbl.loader.DatabankLoader.fetch_databank).

### How to bypass it 

`drop_non_numeric` cannot be changed from `calc_spectrum`. You need to use the lower level SpectrumFactory class : 

This does the same thing as your script. Notice the addition of `drop_non_numeric=False`
```
from radis import SpectrumFactory

sf = SpectrumFactory(wavelength_min=4197,
                         wavelength_max=4254,
                         mole_fraction=0.001,
                         molecule='H2O',
                         isotope='1,2,3,4,5',
                         path_length=1,
                          verbose=3,)
sf.fetch_databank('hitran', drop_non_numeric=False)
sf.eq_spectrum(Tgas=300, pressure=1.01325)

```




### Why did it happen ? 

No idea. 
I printed the value of `id` in all lines above : 

```
sf.df0.id
Out[38]: 
0      1
1      1
2      1
3      1
4      1
      ..
468    1
469    1
470    1
471    1
472    1
Name: id, Length: 473, dtype: object
```

```
(sf.df0.id == 1).all()
>> True
```

Therefore all lines are correctly labelled "1" (no error or missing values in the HITRAN database). But they are not properly converted to numeric values on loading ? 


Possible ways to fix it : 

- prevent drop_non_numeric from dropping `id` and `iso` (tell him to keep the minimal amount of columns).
- totally remove drop_non_numeric.  This is related to the Reduce Memory Usage project #118 (@gagan-aryan you'll be interested) : if optimizing memory, we only need to keep the appropriate columns, not discard the non-numeric ones specifically. 
- or ensure, while loading, that `id` and `iso` are numerical values. ⚠️ this may be required anyway, some later, non-equilibrium functions assume isotopes are integer values. 

In any case, to close this we need to find out why it happens on this specific spectral range and not the others you had tried.

**Comment by CorentinGrimaldi** [comment]
After more investigations, the culprit is the 5th isotope of H2O which has no line between 4197 and 4254 nm thus the error : 'DataFrame' object has no attribute 'id'.
(I tried to cast the 'id' field before then realized the problem)
So a possible fix could be to discard the irrelevant isotope(s) ?

**Comment by erwanp** [comment]
Well done, this one wasn't easy to find !

Easy to fix, though:   add `if len(df) >0` in [fetch_databank #L942](https://github.com/radis/radis/blob/develop/radis/lbl/loader.py#L942)
```
if len(df) > 0: frame.append(frames.append(df))
```

**Comment by CorentinGrimaldi** [comment]
Maybe also, send a warning to the user ? No need to raise an error though, since at least one isotope has lines in the specified wavelength range.

**Comment by erwanp** [comment]
Good idea. A warning ([EmptyDatabaseWarning](https://github.com/radis/radis/blob/674b883c68e82a255cdf0187a3f59dfcbf2b5180/radis/misc/warning.py#L97) ?), or just a print(). 

RADIS already include a way to have different levels of warning, not all of them would be printed for the same verbose level. See [SpectrumFactory.warn](https://github.com/radis/radis/blob/674b883c68e82a255cdf0187a3f59dfcbf2b5180/radis/lbl/loader.py#L2290) . Something like this would work : 

```
if len(df) > 0:
    frame.append(frames.append(df))
else:
    self.warn("No line for isotope", "EmptyDatabaseWarning", level=2)     # only prints if verbose>=2 
```

Although the implementation of warnings may change in the future, it's a good thing not to clutter the console with non relevant messages ! (tagging @dcmvdbekerom , you may be interested in this "level of warnings" feature)

**Comment by CorentinGrimaldi** [comment]
This issue was closed by PR #253
